src/input.rs

```diff
@@ -0,0 +1,232 @@
+use std::collections::HashMap;
+
+use winit::event::{
+    ElementState,
+    Event,
+    KeyboardInput,
+    MouseButton,
+    MouseScrollDelta,
+    VirtualKeyCode,
+    WindowEvent,
+};
+
+// TODO: should use 'Vectors' from other part?
+#[derive(Debug, Copy, Clone)]
+/// Representation of 2D vectors and points
+pub struct Vector2 {
```

Comment:
Should use 'Vectors' from other part of application? Or should we add our implementation? I would like to have Vector2 and Vector3 struct in our app, that will have usefull methods, like for example distance. (https://docs.unity3d.com/ScriptReference/Vector3.html)
Or have som math "class" for that.

Reply:
We are using [cgmath](https://docs.rs/cgmath/0.17.0/cgmath/) library already.
i.e. cgmath::Vector2<f32>, cgmath::Vector3<f32>, cgmath::Vector2<u32>, etc